pkg/certifier/osv/osv.go

```diff
@@ -50,10 +51,14 @@ func NewOSVCertificationParser() certifier.Certifier {
 
 // CertifyComponent takes in the root component from the gauc database and does a recursive scan
 // to generate vulnerability attestations
-func (o *osvCertifier) CertifyComponent(ctx context.Context, rootComponent *certifier.Component, docChannel chan<- *processor.Document) error {
-	o.rootComponents = rootComponent
+func (o *osvCertifier) CertifyComponent(ctx context.Context, rootComponent interface{}, docChannel chan<- *processor.Document) error {
+	if component, ok := rootComponent.(*certifier.Component); ok {
+		o.rootComponents = component
+	} else {
+		return errors.New("rootComponent type is not *certifier.Component")
```

Comment:
Instead of this, can we declare a new error?

Reply:
Changed to `fmt.Errorf`